python/src/uagents/registration.py

```diff
@@ -167,6 +188,41 @@ async def register(
         )
         if success:
             self._logger.info("Registration on Almanac API successful")
+        else:
+            self._logger.warning("Registration on Almanac API failed")
+
+
+class BatchAlmanacApiRegistrationPolicy(AgentRegistrationPolicy):
+    def __init__(
+        self, almanac_api: Optional[str] = None, logger: Optional[logging.Logger] = None
+    ):
+        self._almanac_api = almanac_api or ALMANAC_API_URL
+        self._attestations: List[AgentRegistrationAttestation] = []
+        self._logger = logger or logging.getLogger(__name__)
+
+    def add_agent(self, agent_info: AgentInfo, identity: Identity):
+        attestation = AgentRegistrationAttestation(
+            agent_address=agent_info.agent_address,
+            protocols=list(agent_info.protocols),
+            endpoints=agent_info.endpoints,
+            metadata=coerce_metadata_to_str(agent_info.metadata),
```

Comment:
This should (for now) only include geolocation. See `AlmanacApiRegistrationPolicy.register()`